allennlp/models/reading_comprehension/dialog_qa.py

```diff
@@ -0,0 +1,408 @@
+import logging
+from typing import Any, Dict, List
+import numpy as np
+from overrides import overrides
+import torch
+import torch.nn.functional as F
+from torch.nn.functional import nll_loss
+
+from allennlp.common import squad_eval
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules import Seq2SeqEncoder, TimeDistributed, TextFieldEmbedder
+from allennlp.modules.matrix_attention.linear_matrix_attention import LinearMatrixAttention
+from allennlp.modules.input_variational_dropout import InputVariationalDropout
+from allennlp.nn import InitializerApplicator, util
+from allennlp.training.metrics import Average, BooleanAccuracy, CategoricalAccuracy
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@Model.register("dialog_qa")
+class DialogQA(Model):
```

Comment:
Minor nitpick: We prefer to use [Google's definition of camel case](https://github.com/allenai/allennlp/blob/master/STYLE.md#naming), so this should be `DialogQa`.